src/traces/bar/plot.js

```diff
@@ -32,128 +31,128 @@ var attributes = require('./attributes'),
 var TEXTPAD = 3;
 
 module.exports = function plot(gd, plotinfo, cdbar, barLayer) {
-    var bartraces = makeTraceGroups(gd, plotinfo, cdbar, barLayer, 'trace bars', plotOne);
-
-    // error bars are on the top
-    Registry.getComponentMethod('errorbars', 'plot')(bartraces, plotinfo);
-};
-
-function plotOne(gd, plotinfo, cd, plotGroup) {
     var xa = plotinfo.xaxis;
     var ya = plotinfo.yaxis;
     var fullLayout = gd._fullLayout;
-    var cd0 = cd[0];
-    var t = cd0.t;
-    var trace = cd0.trace;
 
-    if(!plotinfo.isRangePlot) cd0.node3 = plotGroup;
+    var bartraces = Lib.makeTraceGroups(barLayer, cdbar, 'trace bars').each(function(cd) {
```

Comment:
Ha yes. `Lib.makeTraceGroups` looks very similar to `Lib.ensureSingle` which have similar purpose. :ok_hand: